File: scripts/aiModels/diff_Csv.py
#pip install pandas

import pandas as pd

# Load CSV files into DataFrames
df1 = pd.read_csv('first.csv')
df2 = pd.read_csv('second.csv')

# Convert the 'word' column of each DataFrame to a set
words1 = set(df1['word'].dropna().str.strip())
words2 = set(df2['word'].dropna().str.strip())

# Find words present in the first CSV but not in the second
diff_words = words1 - words2

# Convert the resulting set back into a DataFrame
result_df = pd.DataFrame(list(diff_words), columns=['word'])

# Save the results to a new CSV file
result_df.to_csv('difference.csv', index=False)

print("Difference CSV created with", len(result_df), "words.")

File: scripts/aiModels/diff_Csv.js
//npm install csv-parser fs
const fs = require('fs');
const csv = require('csv-parser');

const words1 = new Set();
const words2 = new Set();

// Function to load CSV words into a Set
function loadWords(filePath, wordSet, callback) {
  fs.createReadStream(filePath)
    .pipe(csv())
    .on('data', (row) => {
      if (row.word) {
        wordSet.add(row.word.trim());
      }
    })
    .on('end', () => {
      console.log(`Finished reading ${filePath}`);
      callback();
    });
}

// Load both CSVs, then compute the difference
loadWords('first.csv', words1, () => {
  loadWords('second.csv', words2, () => {
    // Compute difference: words in words1 not in words2
    const difference = [...words1].filter(word => !words2.has(word));

    // Write to a new CSV file
    const output = difference.map(word => ({ word }));
    const csvHeader = 'word\n';
    const csvContent = output.map(obj => obj.word).join('\n');

    fs.writeFile('difference.csv', csvHeader + csvContent, (err) => {
      if (err) {
        console.error('Error writing CSV:', err);
      } else {
        console.log('difference.csv has been created with', difference.length, 'words.');
      }
    });
  });
});

